domain/lib/src/usecases/received/get_all_received_shares_interactor.dart

```diff
@@ -44,8 +46,23 @@ class GetAllReceivedSharesInteractor {
 
   Future<Either<Failure, Success>> execute() async {
     try {
-      final receivedShares = await _receivedShareRepository.getAllReceivedShares();
-      return Right<Failure, Success>(GetAllReceivedShareSuccess(receivedShares));
+      final receivedShares = await _receivedShareRepository.getAllReceivedShares()
+        .onError((error, stackTrace) => _receivedShareRepository.getAllReceivedShareOffline());
+      final combinedReceivedShares = List<ReceivedShare>.empty(growable: true);
+
+      if (receivedShares.isNotEmpty) {
+        for (final received in receivedShares) {
+          final localReceivedShare = await _receivedShareRepository
```

Comment:
Why don't you use `stream` here and do `combine` it in the `repository`?

Reply:
can you more details?